### ISSUE TITLE
DWG: Assign bags to buckets

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/joystream#4725]
@traumschule commented on [Wed Mar 29 2023](https://github.com/Joystream/pioneer/issues/4306)

As DWG lead i want to be able to assign non-empty channels to different buckets.

The UI should be able to show size for each bag (channel) and deselect some by clicking on it.
- show all buckets
- expandable list of channels for each

Because there are ten thousands of channels the UI needs to provide Buttons for shortcuts per bucket like 'Select all', 'Select none', 'Select non-empty' or accept a list of bag IDs (input field).

To apply selected changes after configuring all buckets lead can click "Apply". The new settings will be applied signing a batch tx (split into multiple blocks of one is not enough).

Query
```
query { channels (limit: 50000,orderBy:createdAt_ASC, where: { videos_some: {isPublic_eq: true} } ) {
  id videos {id}
}}
```

Relevant [JSS code](https://git.l1.media/JoystreamStats/jsstats/src/frontend/src/components/Distribution) and design:

![buckets](https://user-images.githubusercontent.com/31551045/228622038-eaf79feb-b88e-456c-8e02-d6eb23e957b0.png)

[Context](https://discord.com/channels/811216481340751934/933726271832227911/1090643290958872646)

---

@dmtrjsg commented on [Mon Apr 03 2023](https://github.com/Joystream/pioneer/issues/4306#issuecomment-1494328080)

Looks like this was meant to go to Joystream repo.. moving issue there 

cc @traumschule

**Comment by bedeho** [comment]
Sorry what UI and product is this in reference to?

**Comment by traumschule** [comment]
> Looks like this was meant to go to Joystream repo.. moving issue there

Actually thought about Pioneer and was curious if you agree that it's relevant for governance. Would require changes to the DWG WG page.

**Comment by bedeho** [comment]
> Actually thought about Pioneer and was curious if you agree that it's relevant for governance. Would require changes to the DWG WG page.

I think its too operational, I think if we try to make Pioneer a working environment for all working groups it will make the product very slow moving and bloated for everyone. The problems you are describing are very specific to the lead, and perhaps - although not obviously, some of the workers, possibly some app operators.

Seems much more appropriate to make a distinct experience for this, and the print screen above seems to indicate that already was the case.

**Comment by traumschule** [comment]
It would make the product more useful for leads. But it's your product.

**Comment by bedeho** [comment]
> It would make the product more useful for leads

Yes, it would make it more useful to the handful of individuals at any given time, but
a) it would not actually be better for those people long term ,because the ability to update and maintain this feature will get way behind their needs.
b) its not clear that having a standalone product for this would actually be worse than having it in pioneer even for these specific individuals
c) it would harm product for everyone else over time on net who derives no direct benefit, by bloating, making it harder to maintain and improve: all new functionality does.

**Comment by bedeho** [comment]
At any rate, moving this to Pioneer repo.
